### PR TITLE
Cache keystone settings per instance not per barclamp

### PR DESCRIPTION
**Pull request by dirkmueller**
per barclamp is actually unnecessarily conservative, per keystone
instance is enough. even in the dumb nova-compute node case this
reduces cache misses by factor two.

### CODE REVIEW

File: chef/cookbooks/keystone/libraries/helpers.rb [hunk 1]
```diff
@@ -90,18 +92,14 @@ def self.keystone_settings(current_node, cookbook_name)
...
+    @keystone_settings[instance].merge(
+      service_user: current_node[cookbook_name][:service_user],
+      service_password: current_node[cookbook_name][:service_password])
   end
 
-  private
-  def self.search_for_keystone(node, cookbook_name)
-    instance = node[cookbook_name][:keystone_instance] || "default"
-
+  private_class_method def self.search_for_keystone(node, instance)
```
Comment (aspiers) [inline chef/cookbooks/keystone/libraries/helpers.rb]: I don't get this change but I'll take it on trust it's right :-)

Comment (dirkmueller) [inline chef/cookbooks/keystone/libraries/helpers.rb]: Thats what houndci told me. apparently ```private:``` does not apply to class methods. some ruby crap..

**Comment by dirkmueller** [comment]
so this doesn#t work because now we're caching so good that the values that are cached are incorrect (because on first run they're not yet set). requires some more cleanup..

File: chef/cookbooks/keystone/libraries/helpers.rb (hunk 1 above)
Comment (rsalevsky) [inline chef/cookbooks/keystone/libraries/helpers.rb]: http://www.rubydoc.info/github/bbatsov/RuboCop/RuboCop/Cop/Lint/IneffectiveAccessModifier

**Pull request merged**